service/frontend/workflowHandler.go

```diff
@@ -701,13 +710,20 @@ func (wh *WorkflowHandler) PollForActivityTask(
 	}
 
 	wh.Service.GetLogger().Debug("Received PollForActivityTask")
-	if pollRequest.Domain == nil {
+	if pollRequest.Domain == nil || pollRequest.GetDomain() == "" {
 		return nil, wh.error(errDomainNotSet, scope)
 	}
 
+	if len(pollRequest.GetDomain()) > common.MaxIDLengthLimit {
+		return nil, wh.error(errDomainTooLong, scope)
+	}
+
 	if err := wh.validateTaskList(pollRequest.TaskList, scope); err != nil {
 		return nil, err
 	}
+	if len(pollRequest.GetIdentity()) > common.MaxIDLengthLimit {
+		return nil, wh.error(errIdentityTooLong, scope)
```

Comment:
Let's make sure we log enough information like DomainName, WorkflowID, RunID, etc when logging these errors.

Reply:
https://github.com/uber/cadence/issues/1364